storage/objects/main.go

```diff
@@ -0,0 +1,196 @@
+	}
+
+	newName, err := move(client, bucketName, objectName)
+	if err != nil {
+		log.Fatalf("failed to move the object: %v", err)
+	}
+	objectName = newName
+
+	if err := delete(client, bucketName, objectName); err != nil {
+		log.Fatalf("failed to delete the object: %v", err)
+	}
+}
+
+func write(client *storage.Client, bucket, object string) error {
+	ctx := context.Background()
+	// [START upload_file]
+	f, err := os.Open("notes.txt")
+	if err != nil {
+		return err
+	}
+	wc := client.Bucket(bucket).Object(object).NewWriter(ctx)
+	if _, err = io.Copy(wc, f); err != nil {
+		return err
+	}
+	if err := f.Close(); err != nil {
+		return err
+	}
+	if err := wc.Close(); err != nil {
+		return err
+	}
+	// [END upload_file]
+	return nil
+}
+
+func read(client *storage.Client, bucket, object string) ([]byte, error) {
+	ctx := context.Background()
+	// [START download_file]
+	rc, err := client.Bucket(bucket).Object(object).NewReader(ctx)
+	if err != nil {
+		return nil, err
+	}
+	data, err := ioutil.ReadAll(rc)
+	if err != nil {
+		return nil, err
+	}
+	if err := rc.Close(); err != nil {
+		return nil, err
+	}
+	return data, nil
+	// [END download_file]
+}
+
+func attrs(client *storage.Client, bucket, object string) (*storage.ObjectAttrs, error) {
+	ctx := context.Background()
+	// [START get_metadata]
+	o := client.Bucket(bucket).Object(object)
+	attrs, err := o.Attrs(ctx)
+	if err != nil {
+		return nil, err
+	}
+	return attrs, nil
+	// [END get_metadata]
+}
+
+func makePublic(client *storage.Client, bucket, object string) error {
+	ctx := context.Background()
+	// [START public]
+	acl := client.Bucket(bucket).Object(object).ACL()
+	if err := acl.Set(ctx, storage.AllUsers, storage.RoleReader); err != nil {
+		return err
+	}
+	// [END public]
+	return nil
+}
+
+func move(client *storage.Client, bucket, object string) (string, error) {
+	ctx := context.Background()
+	// [START move_file]
+	dstName := object + "-rename"
+
+	src := client.Bucket(bucket).Object(object)
+	dst := client.Bucket(bucket).Object(dstName)
+
+	if _, err := src.CopyTo(ctx, dst, nil); err != nil {
+		return "", err
+	}
+	if err := src.Delete(ctx); err != nil {
+		return "", err
+	}
+	// [END move_file]
+	return dstName, nil
+}
+
+func copyToBucket(client *storage.Client, dstBucket, srcBucket, srcObject string) error {
+	ctx := context.Background()
+	// [START copy_file]
+	src := client.Bucket(srcBucket).Object(srcObject)
+	dst := client.Bucket(dstBucket).Object(srcObject + "-copy")
+
+	if _, err := src.CopyTo(ctx, dst, nil); err != nil {
+		return err
+	}
+	// [END copy_file]
+	return nil
+}
+
+func delete(client *storage.Client, bucket, object string) error {
+	ctx := context.Background()
+	// [START download_file]
+	o := client.Bucket(bucket).Object(object)
+	if err := o.Delete(ctx); err != nil {
+		return err
+	}
+	// [END download_file]
+	return nil
+}
+
+func createBucketIfNotexits(ctx context.Context, client *storage.Client, projectID, bucket string) error {
```

Comment:
I don't think the motivation behind these samples is to distribute them as actual tools. If so, they are terribly structured and poorly documented. I assume, we won't use any of these other than providing snippets. And generating a bucket automatically makes this a non-tool.